src/pycram/process_modules/icub_process_modules.py

```diff
@@ -0,0 +1,687 @@
+
+    def looking(self):
+        if ProcessModuleManager.execution_type == ExecutionType.SIMULATED:
+            return iCubMoveHead(self._looking_lock)
+        elif ProcessModuleManager.execution_type == ExecutionType.REAL:
+            return iCubMoveHeadReal(self._looking_lock,self.gaze_client_port)
+
+    def detecting(self):
+        if ProcessModuleManager.execution_type == ExecutionType.SIMULATED:
+            return iCubDetecting(self._detecting_lock)
+        elif ProcessModuleManager.execution_type == ExecutionType.REAL:
+            return iCubDetectingReal(self._detecting_lock)
+
+    def move_tcp(self):
+        if ProcessModuleManager.execution_type == ExecutionType.SIMULATED:
+            return iCubMoveTCP(self._move_tcp_lock)
+        elif ProcessModuleManager.execution_type == ExecutionType.REAL:
+            return iCubMoveTCPReal(self._move_tcp_lock,self.action_client_port)
+
+    def move_arm_joints(self):
+        if ProcessModuleManager.execution_type == ExecutionType.SIMULATED:
+            return iCubMoveArmJoints(self._move_arm_joints_lock)
+
+        elif ProcessModuleManager.execution_type == ExecutionType.REAL:
+            return iCubMoveArmJointsReal(self._move_arm_joints_lock,
+                                     [self.state_right_arm_port, self.state_left_arm_port],
+                                     [self.ctp_right_arm_client_port,self.ctp_left_arm_client_port])
+
+    def world_state_detecting(self):
+        if (ProcessModuleManager.execution_type == ExecutionType.SIMULATED or
+                ProcessModuleManager.execution_type == ExecutionType.REAL):
+            return iCubWorldStateDetecting(self._world_state_detecting_lock)
+
+    def move_joints(self):
+        if ProcessModuleManager.execution_type == ExecutionType.SIMULATED:
+            return iCubMoveJoints(self._move_joints_lock)
+
+        elif ProcessModuleManager.execution_type == ExecutionType.REAL:
+            return iCubMoveJointsReal(self._move_joints_lock,
+                                  [self.state_torso_port,self.state_right_arm_port,self.state_left_arm_port],
+                                  [self.ctp_torso_client_port,self.ctp_right_arm_client_port,self.ctp_left_arm_client_port])
+
+    def move_gripper(self):
+        if ProcessModuleManager.execution_type == ExecutionType.SIMULATED:
+            return iCubMoveGripper(self._move_gripper_lock)
+        elif ProcessModuleManager.execution_type == ExecutionType.REAL:
+            return iCubMoveGripperReal(self._move_gripper_lock,
+                                     [self.state_right_arm_port, self.state_left_arm_port],
+                                     [self.ctp_right_arm_client_port,self.ctp_left_arm_client_port])
+
+    def open(self):
+        print("iCub doesn't perform open action from here")
+
+    def close(self):
+        print("iCub doesn't perform close action from here")
+
+
+    def config_yarp_ports(self)->bool:
```

Comment:
the implementation of config methods should also go into an interface file to yarp instead of the process module manager and the just called from here

Reply:
In this configuration function, I open and connect specific ports to control and read the status of the robot. It doesn't configure the network or general yarp things. It is more related to the direct low-level control of the robot. 

IIf you still think that it should be somehow in the external_interface, I can try to restructure the code for that. 
But it is kinda a major change. Because it will change the general pipeline of the code and its initialization